source/network/network.go

```diff
@@ -112,29 +116,30 @@ func (s *networkSource) GetFeatures() *nfdv1alpha1.Features {
 	return s.features
 }
 
-func detectNetDevices() ([]nfdv1alpha1.InstanceFeature, error) {
+func detectNetIfaces() ([]nfdv1alpha1.InstanceFeature, []nfdv1alpha1.InstanceFeature, error) {
 	sysfsBasePath := hostpath.SysfsDir.Path(sysfsBaseDir)
 
 	ifaces, err := os.ReadDir(sysfsBasePath)
 	if err != nil {
-		return nil, fmt.Errorf("failed to list network interfaces: %w", err)
+		return nil, nil, fmt.Errorf("failed to list network interfaces: %w", err)
 	}
 
-	// Iterate over devices
-	info := make([]nfdv1alpha1.InstanceFeature, 0, len(ifaces))
+	// Iterate over interfaces
+	virtualIfacesinfo := make([]nfdv1alpha1.InstanceFeature, 0, len(ifaces))
+	deviceIfacesinfo := make([]nfdv1alpha1.InstanceFeature, 0, len(ifaces))
 	for _, iface := range ifaces {
 		name := iface.Name()
 		if _, err := os.Stat(filepath.Join(sysfsBasePath, name, "device")); err == nil {
-			info = append(info, readIfaceInfo(filepath.Join(sysfsBasePath, name)))
-		} else if klogV := klog.V(3); klogV.Enabled() {
-			klogV.InfoS("skipping non-device iface", "interfaceName", name)
+			virtualIfacesinfo = append(virtualIfacesinfo, readIfaceInfo(filepath.Join(sysfsBasePath, name), true))
+		} else {
+			deviceIfacesinfo = append(deviceIfacesinfo, readIfaceInfo(filepath.Join(sysfsBasePath, name), false))
 		}
 	}
 
-	return info, nil
+	return virtualIfacesinfo, deviceIfacesinfo, nil
 }
 
-func readIfaceInfo(path string) nfdv1alpha1.InstanceFeature {
+func readIfaceInfo(path string, isVirtual bool) nfdv1alpha1.InstanceFeature {
```

Comment:
Looking at this I feel it would be better to refactor the function in a more generic format. I submitted my suggestion as a PR, see #1452.

Also, you need to have a different set of attributes for the virtual interfaces, e.g. I don't see `speed` attribute added in this PR